### DISCUSSION TITLE
Seeking guidance on implementing a proxy in Ruby

### DISCUSSION THREAD

**Discussion by muldersjoost80** [gorilla/.github#30] [Q&A] [answered]
I'm currently engaged in a project (new ver of https://proxymimarlik.com/) where I'm aiming to implement a proxy server using Ruby. My primary objective is to create a proxy that can efficiently handle HTTP requests and responses. However, I'm encountering some challenges, and I'm looking for guidance or resources that could help.

Here are the specifics of what I'm trying to achieve:

**Functionality:** The proxy needs to intercept, analyze, and possibly modify HTTP requests and responses.

**Performance:** It should be optimized for performance, able to handle a high volume of traffic.

**Security:** Implementing essential security features to prevent common vulnerabilities.

I have the following questions:

1. Are there any existing libraries or frameworks in Ruby that could simplify this process?
2. What are the best practices for ensuring high performance and security in such a proxy server?
3. Could anyone share sample code or a repository that demonstrates a similar implementation?

**Comment by jaitaiwan** [comment, marked as answer]
I’m not sure you’ll find much help here. This is a golang project.
